### PR TITLE
Reworded the explanation about optional command options

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | 2.3+ |
| Fixed tickets | #4110 |

### CODE REVIEW

File: components/console/introduction.rst [hunk 1]
```diff
@@ -296,9 +296,12 @@ declare a one-letter shortcut that you can call with a single dash like
 
 .. tip::
 
-    It is also possible to make an option *optionally* accept a value (so that
-    ``--yell``, ``--yell=loud`` or ``--yell loud`` work). Options can also be configured to
-    accept an array of values.
+    It may look that it's possible to make an option *optionally* accept a value
```
Comment (cordoval) [inline components/console/introduction.rst]: the option optionally here sounds very confusing, can it be reworded too?

Comment (WouterJ) [inline components/console/introduction.rst]: I can't come up with anything else... (btw, you're back reviewing, awesome!).

Comment (cordoval) [inline components/console/introduction.rst]: We can remove it all together and say:

```
There is nothing forbidding you to create a command with an input option set to InputOption::OPTIONAL mode. You will just not be able to distinguish when the flag for that option was used without a value (``command --yell``) or when it was not used at all (``command``). The value retrieved for the option will be the same ``null``.
```
?

Comment (javiereguiluz) [inline components/console/introduction.rst]: I like your explanation ... but I've tweaked some minor parts of it. Thanks.

File: components/console/introduction.rst [hunk 2]
```diff
@@ -296,9 +296,11 @@ declare a one-letter shortcut that you can call with a single dash like
 
 .. tip::
 
-    It is also possible to make an option *optionally* accept a value (so that
-    ``--yell``, ``--yell=loud`` or ``--yell loud`` work). Options can also be configured to
-    accept an array of values.
+    There is nothing forbidding you to create a command with an *optional option*,
```
Comment (WouterJ) [inline components/console/introduction.rst]: -1 on the term "optional option", options are always optional. It's the value we are talking about here

File: components/console/introduction.rst [hunk 3]
```diff
@@ -296,9 +296,11 @@ declare a one-letter shortcut that you can call with a single dash like
...
+    which is an input option set to ``InputOption::OPTIONAL`` mode. You will just
+    not be able to distinguish when the flag for that option was used without a
+    value (``command --yell``) or when it was not used at all (``command``).
+    In both cases the value retrieved for the option will be the same ``null``.
```
Comment (WouterJ) [inline components/console/introduction.rst]: Let me try again :)

 > There is nothing forbidding you to create a command with an option that optionally accepts a value. However, there is no way you can distinguish when the option was used without a values (``command --yell``) or when it wasn't used at all (``command``). In both cases, the value retrieved for the option will be ``null``.

Comment (xabbuh) [inline components/console/introduction.rst]: :+1: (but with "without a value")

Comment (javiereguiluz) [inline components/console/introduction.rst]: Thanks for your suggestions. It's much better now.

**Comment by xabbuh** [comment]
Looks good :+1:

File: components/console/introduction.rst [hunk 4]
```diff
@@ -296,9 +296,11 @@ declare a one-letter shortcut that you can call with a single dash like
 
 .. tip::
 
-    It is also possible to make an option *optionally* accept a value (so that
-    ``--yell``, ``--yell=loud`` or ``--yell loud`` work). Options can also be configured to
-    accept an array of values.
+    There is nothing forbidding you to create a command with an option that
```
Comment (cordoval) [inline components/console/introduction.rst]: that optionally again :older_man:, it is okay, i will not fuss again about that. Go ahead :+1:

**Comment by weaverryan** [comment]
:thumbsup: Thanks Javier :)
